### PR TITLE
Prevent adding NaN values to the list

### PR DESCRIPTION
**Pull request by andrepiske**
1. Prevent NaN values from being added to the list. This is achieved by disabling the "Save" button only
2. Change "i.e." to "e.g." in the input field

My CSS additions looks bad. Sorry, I've never used Stylus. Could you please suggest improvements to it?

### CODE REVIEW

File: src/styl/_native.styl [hunk 1]
```diff
@@ -31,5 +31,12 @@ button
...
     border 1px solid $color-primary
+    &.disabled
+      background-color $color-grey
+      border-color: $color-grey
     &:hover
       background lighten($color-primary, 5%)
+      &.disabled
+        background-color $color-grey
+        border-color $color-grey
+        cursor not-allowed
```
Comment (rafaelcamargo) [inline src/styl/_native.styl]: What does make `not-allowed`? I usually use `default`.

File: src/js/popup.js [hunk 2]
```diff
@@ -23,14 +24,27 @@
...
   function onButtonSaveClick(){
     var USDEnteredValue = getUSDValueEntered();
     addUSDValueToAlertsList(USDEnteredValue);
     chromeService.storage.addAlert(USDEnteredValue);
   }
 
   function getUSDValueEntered(){
-    return parseFloat($('[data-js="usd-value"]').val().replace(',','.'));
+    var enteredValue = $('[data-js="usd-value"]').val().replace(',', '.');
+    if (!enteredValue.match(/^[0-9]+(\.[0-9]+)?$/))
```
Comment (rafaelcamargo) [inline src/js/popup.js]: Things can be better described as follows:

``` javascript
function onUSDValueChange(){
  var buttonSave = $('[data-js="button-save"]');
  var USDEnteredValue = getUSDValueEntered();
  if(USDEnteredValue)
    setupSaveButtonAbility(buttonSave, 'enable');
  else
    setupSaveButtonAbility(buttonSave, 'disable');
}

function getUSDValueEntered(){
  var enteredValue = $('[data-js="usd-value"]').val().replace(',', '.');
  if(isUSDValueEnteredValid(enteredValue))
    return parseFloat(enteredValue);
}

function isUSDValueEnteredValid(enteredValue){
  return enteredValue.match(/^[0-9]+(\.[0-9]+)?$/);
}

function setupSaveButtonAbility(buttonSave, behaviour){
  buttonSave.addClass('disabled').prop('disabled', true);
  if(behaviour == 'enable')
    buttonSave.removeClass('disabled').prop('disabled', false);
}
```

File: src/jade/popup.jade [hunk 3]
```diff
@@ -10,8 +10,8 @@ html(class="dollert-popup")
         img(src="logotype.png")
       .dollert-alert-form-label Enter the USD value you are looking for in BRL:
       .dollert-input-buttoned
-        input(data-js="usd-value" placeholder="i.e: 3.70")
-        button.is-primary(data-js="button-save" type="button") Save
+        input(data-js="usd-value" placeholder="e.g: 3.70")
+        button.is-primary(data-js="button-save" type="button").disabled(disabled) Save
```
Comment (rafaelcamargo) [inline src/jade/popup.jade]: It is better to keep css classes and html attributes grouped:
`button.is-primary.disabled(data-js="button-save" type="button" disabled)`

File: src/jade/popup.jade [hunk 4]
```diff
@@ -10,8 +10,8 @@ html(class="dollert-popup")
         img(src="logotype.png")
       .dollert-alert-form-label Enter the USD value you are looking for in BRL:
       .dollert-input-buttoned
-        input(data-js="usd-value" placeholder="i.e: 3.70")
-        button.is-primary(data-js="button-save" type="button") Save
+        input(data-js="usd-value" placeholder="e.g: 3.70")
```
Comment (rafaelcamargo) [inline src/jade/popup.jade]: 🍻

File: src/js/popup.js [hunk 5]
```diff
@@ -10,6 +10,7 @@
 
   function bindElements(){
     $('[data-js="button-save"]').on('click', onButtonSaveClick);
+    $('[data-js="usd-value"]').on('change paste keyup', onUSDValueChange);
```
Comment (rafaelcamargo) [inline src/js/popup.js]: Wouldn't be only `keyup` enough?

**Comment by andrepiske** [comment]
@rafaelcamargo What about this?

**Comment by rafaelcamargo** [comment]
@andrepiske That was such readability I was previously mean (very small functions with isolated and well defined responsibility): https://github.com/rafaelcamargo/dollert/pull/46/commits/8beb2e51905aaf482244bcf5d9741a0d51ff535d

**Pull request closed without merging**